packages/nodejs/scripts/extension/extension.js

```diff
@@ -47,36 +48,40 @@ function downloadFromMirror(mirror, filename, outputPath) {
 
     const url = [mirror, AGENT_VERSION, filename].join("/")
 
-    https
-      .get(url, response => {
-        const { statusCode } = response
+    const transmitter = new Transmitter(url)
 
-        if (statusCode >= 400) {
-          reject(
-            new DownloadError(
-              `Request to CDN failed with code HTTP ${statusCode}`,
-              url
+    const promise = transmitter.download()
```

Comment:
```suggestion
    const downloadPromise = transmitter.download()
```

What about renaming this one?

Reply:
Ended up getting rid of this `const` declaration entirely, to avoid giving this a name at all. 🎉